geographic_msgs/CMakeLists.txt

```diff
@@ -40,11 +39,35 @@ rosidl_generate_interfaces(${PROJECT_NAME}
   DEPENDENCIES unique_identifier_msgs geometry_msgs std_msgs builtin_interfaces
 )
 
+add_library(validation INTERFACE)
+add_library(${PROJECT_NAME}::validation ALIAS validation)
+include(GNUInstallDirs)
+add_subdirectory(include)
+
+include(CTest)
+if(BUILD_TESTING)
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
+  add_subdirectory(test)
+endif()
+
 install(
   FILES geographic_msgs_mapping_rule.yaml
   DESTINATION share/${PROJECT_NAME}
 )
 
+install(
+  TARGETS validation
+  EXPORT export_${PROJECT_NAME}
+)
+
+install(
+  DIRECTORY include/
+  DESTINATION include/${PROJECT_NAME}
+  FILES_MATCHING PATTERN "*.hpp"
```

Comment:
I added a matcher to not install the CMakeListst.txt in the installation directory.